session/src/main/java/org/apache/iotdb/session/Session.java

```diff
@@ -654,8 +648,7 @@ public void executeNonQueryStatement(String sql) throws TException, IoTDBRPCExce
   private void checkPathValidity(String path) throws IoTDBSessionException {
     if (!Pattern.matches(PATH_MATCHER, path)) {
```

Comment:
If a pattern is to be used multiple times, compiling it once and reusing it will be more efficient than invoking this method each time.

Reply:
Hi, thanks a lot for this suggestion! That's quite useful, and I have updated it in the latest commit. : )